camel/agents/role_playing.py

```diff
@@ -25,6 +26,22 @@
 from camel.typing import ModelType, RoleType, TaskType
 
 
+@dataclass(frozen=True)
+class MessageWithExtras:
+    r"""ChatMessage grouped with extra info.
+
+    Args:
+        msg (Optional[ChatMessage]): ChatMessage obtained from a chat agent.
+        terminated (Optional[bool]): Whether the agent requested termination
+            of the chat session.
+        info (Optional[Dict[str, Any]]): Extrainformation aasciated with
+            the current message.
+    """
+    msg: Optional[ChatMessage]
+    terminated: Optional[bool]
+    info: Optional[Dict[str, Any]]
+
+
```

Comment:
Why it is essential to have both `ChatAgentResponse` and `MessageWithExtras`?

Reply:
Yeah, let's merge them

Reply:
Done